### DISCUSSION TITLE
How do i prevent auto disconnect?

### DISCUSSION THREAD

**Discussion by Santosl2** [socketio/socket.io#4739] [Q&A]
I have a application built in React Native (Mobile) and Node js (Backend). When user is in my app and goes to another app and stays in this app for few minutes, my socket is automatically disconnected.

I saw some 'fix' increasing pingTimeout and pingInterval, is it a solution or is there another solution?

**Comment by Santosl2** [comment]
someone ?

**Comment by yeuci** [comment]
yes, that is a solution.

ping timeout
`js
io.set('heartbeat timeout', 60000);
`

the timeout for the initial connection handshake
`js
io.set('connect timeout', 10000);`

changing these can give you a longer period of inactivity

**Reply by yeuci** [reply]
link to the docs - https://socket.io/docs/v4/how-it-works/#:~:text=There%20is%20also%20a%20heartbeat,send%20a%20PONG%20packet%20back.

**Comment by darrachequesne** [comment]
Hi!

> When user is in my app and goes to another app and stays in this app for few minutes, my socket is automatically disconnected.

I think that's an expected behavior, the application is put in background and all connections are closed to save battery.

References:

- https://socket.io/docs/v4/#what-socketio-is-not
- https://socket.io/docs/v4/troubleshooting-connection-issues/#problem-the-socket-gets-disconnected

**Reply by Santosl2** [reply]
do you know another way to prevent this?

**Reply by darrachequesne** [reply]
I don't think you should prevent this. As noted [here](https://socket.io/docs/v4/#what-socketio-is-not), keeping a WebSocket connection open in the background would quickly drain the battery of your users.

**Reply by Santosl2** [reply]
hmmm i'll try to do this. But very strange

**Comment by anxheloo** [comment]
Hello guys, im having same problem. Im working on a react native chat app and when the app goes inactive or go into another app without closing it, the socket connection is closed. Than if other user sends a message, and i come back to AppState active, the message is not shown. How to deal with this case, any suggestion? Thanks in advance
